src/api/wasi/state.rs

```diff
@@ -133,24 +143,33 @@ impl WasiState {
 
     pub fn filestat_path<P: AsRef<Path>>(&self, abs_path: P) -> Result<Filestat, Status> {
         let metadata = fs::metadata(&abs_path)?;
+        // NOTE try_into cast should work correctly on glibc systems as tv_nsec will be less
```

Comment:
@akegalj so `tv_nsec` is `u32` less than 1e9, but even if it was 2^32 - 1, this would work hm They simply note that it's less than 1e9 because that's a whole second. Anyway, I think this is fine.